discretisedfield/tests/test_field.py

```diff
@@ -3151,14 +3151,15 @@ def test_mpl(test_field):
     plt.close("all")
 
 
-def test_mpl_dimension(valid_mesh):
-    field = df.Field(valid_mesh, nvdim=1)
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+def test_mpl_dimension(valid_mesh, nvdim):
+    field = df.Field(valid_mesh, nvdim=nvdim)
 
-    if valid_mesh.region.ndim != 2:
+    if valid_mesh.region.ndim != 2 or nvdim not in [1, 2, 3]:
         with pytest.raises(RuntimeError):
-            field.mpl.scalar()
+            field.mpl()
```

Comment:
For a wrong number of `vdims` we get a different error:
```python
>>> (field.sel(z=10e-9) << field.sel(z=10e-9)).mpl()
...
ValueError: At least one element of vdims=[None, None] must be not None.
```

Side note, not to be added to the test:
Plotting would still work for higher-dimensional fields when we specify `vdims`:
```python
(field.sel(z=10e-9) << field.sel(z=10e-9)).mpl(vector_kw={'vdims': ['v2', 'v5']})
```
Only a subset of 3 vdims is then shown in the plot.